### ISSUE TITLE
WIP: Undersøgelse af lasses overlegende performance

### ISSUE THREAD

**Issue by AndreasMadsen** [AndreasMadsen/course-02460#5]
lasse: https://github.com/AndreasMadsen/course-02460/tree/80995b843952af77a2c5dea7933dab9a6854a8c4
andreas: https://github.com/AndreasMadsen/course-02460/tree/323afbab1120b1cea5645c5e242a7133d396af34
## Lasses test loss

Lasses test loss er udregnet på træningssættet:

https://github.com/AndreasMadsen/course-02460/blob/80995b843952af77a2c5dea7933dab9a6854a8c4/code/main.py#L74L78
## `num_filters` i `Conv2D`

lasse bruger: 16
andreas bruger: 32

`num_filters=32` giver nan værdier for både (`nfft=256`, `truncate=300`) og (`nfft=512`, `truncate=150`) i andreas implementering. Men i lasses implementering giver `num_filters=32` ikke nan værdier. Der må altså være en 3. faktor.

`num_filters=16` giver ikke nan værdier for begge `nfft` tilfælde.
## `nfft` i spectogram

lasse bruger: `nfft=256`, `truncate=300`
andreas bruger: `nfft=512`, `truncate=150`

![skaermbillede 2016-03-11 kl 14 38 41](https://cloud.githubusercontent.com/assets/505333/13703526/0c0e9936-e797-11e5-9685-85cc83e44c99.png)
_Figure ^: `nfft = 256` og `truncate=300`, `filters=16`_

![skaermbillede 2016-03-11 kl 14 43 23](https://cloud.githubusercontent.com/assets/505333/13703622/abd48dea-e797-11e5-8671-20e8ccec7d4e.png)
_Figure ^: `nfft = 512` og `truncate=150`, `filters=16`_

det ser ud til at `256` giver en meget hurtigere learning rate men overfitter også langt hurtigere. Lasses når dog ned på `0.08` loss efter 100 epochs, så der bør være andre faktorer.
## dataset selektion (`noverlap`)

Ved at bruge andreas dataset selektor med lasses network builder, ses det at `num_filters=32` giver nan og `num_filters=16` ikke gør. Dette udelukker umiddelbart netværks implementering som faktor.

lasse bruger: `usage='train'`, `dialect='dr1'`, `batch_size=50`, `shuffle=True`
andreas bruger: `usage='train'`, `dialect='dr1'`, `batch_size=50`, `shuffle=True`

så den umiddelbare forskel på API niveauet er ikke til at se.

Lasses data filter giver `285` resultater, andreas's data filter giver `41` resultater. Der er `380` `_.wav` filer i `dr1` mappen, lasses og andreas's data selector giver `380` filer. Ved at fjerne `noverlap=128` fra https://github.com/AndreasMadsen/course-02460/blob/master/code/load_data.py#L175 giver lasses filter også `41` resultater.

![skaermbillede 2016-03-11 kl 17 29 36](https://cloud.githubusercontent.com/assets/505333/13708431/e445c074-e7ae-11e5-9147-68795450f67a.png)

_Figure: ^: bruger samme netværks kode som lasse og andreas's data filter er ændret så den opføre sig som lasses._

Nu fås den samme gode performance som lasse.
